### DISCUSSION TITLE
How to integrate seperate create react app to a Laravel backend server?

### DISCUSSION THREAD

**Discussion by thien1103** [community/community#133605] [Programming Help]
### Select Topic Area

Question

### Body

Hi guys, is there anyway to connect my react app to a Laravel backend server which already set a mixpack (now I don't want to use the integrate React anymore). Any support from you guys will be appreciated. Thanks for reading!

**Comment by KuyaMecky** [comment]
Yes, you can definitely connect a separate React app to a Laravel backend server. Since you want to keep the React app and Laravel backend as separate entities, you’ll need to set up your React app to communicate with your Laravel API. Here’s a high-level overview of the steps:

### 1. **Set Up Laravel Backend**

1. **Create API Routes**: Define routes in Laravel that will serve your API endpoints. You’ll typically do this in the `routes/api.php` file. For example:

   ```php
   Route::get('/users', 'UserController@index');
   Route::post('/users', 'UserController@store');
   ```

2. **Configure CORS**: Ensure that Laravel can handle requests from your React app by configuring CORS. You can do this by adding the `fruitcake/laravel-cors` package or configuring CORS in `config/cors.php`:

   ```php
   'supports_credentials' => true,
   'allowed_origins' => ['http://localhost:3000'],
   ```

3. **Test API Endpoints**: Use tools like Postman or CURL to ensure that your API endpoints are working as expected.

### 2. **Set Up React Frontend**

1. **Install Axios or Fetch**: Use Axios or the native Fetch API to make HTTP requests to your Laravel API. First, install Axios if you choose to use it:

   ```bash
   npm install axios
   ```

2. **Configure Requests**: In your React components, make requests to the Laravel API endpoints. For example, using Axios:

   ```javascript
   import axios from 'axios';

   axios.get('http://localhost:8000/api/users')
     .then(response => {
       console.log(response.data);
     })
     .catch(error => {
       console.error('There was an error!', error);
     });
   ```

   Or using Fetch:

   ```javascript
   fetch('http://localhost:8000/api/users')
     .then(response => response.json())
     .then(data => console.log(data))
     .catch(error => console.error('Error:', error));
   ```

3. **Handle Environment Variables**: For managing API base URLs and other sensitive information, use environment variables. Create a `.env` file in your React project and add:

   ```env
   REACT_APP_API_URL=http://localhost:8000/api
   ```

   Then access it in your code:

   ```javascript
   const apiUrl = process.env.REACT_APP_API_URL;
   ```

### 3. **Run and Test**

1. **Start Laravel Server**: Run your Laravel backend server using:

   ```bash
   php artisan serve
   ```

2. **Start React App**: Run your React app using:

   ```bash
   npm start
   ```

3. **Test Integration**: Navigate to your React app and test if it successfully communicates with your Laravel API.

By following these steps, you should be able to integrate your separate React app with your Laravel backend server efficiently. If you run into any specific issues, feel free to ask for more help!

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @thien1103 ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.
